lib/auth/middleware.go

```diff
@@ -63,6 +65,8 @@ type TLSServerConfig struct {
 	AcceptedUsage []string
 	// ID is an optional debugging ID
 	ID string
+	// Metrics are optional TLSServer metrics
+	Metrics *Metrics
```

Comment:
If this wasn't a pointer you could rely on default values to do the right thing at the moment since `Metrics` only contains one boolean.  If `Metrics` was to be extended in the future then relying on default values wouldn't work so maybe it's fine to leave as it is.